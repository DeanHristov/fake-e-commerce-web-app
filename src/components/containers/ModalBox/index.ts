import ModalBox, { IModalBoxProps } from './ModalBox';

export type { IModalBoxProps };
export default ModalBox;
