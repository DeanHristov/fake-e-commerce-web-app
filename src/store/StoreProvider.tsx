'use client';

/* Core */
import { Provider } from 'react-redux'; /* Instruments */
import { reduxStore } from './';
import { ReactNode } from 'react';

interface IProvidersProps {
  children: ReactNode;
}

// Including mocks
const isMocking: boolean =
  process.env.NODE_ENV !== 'test' &&
  Boolean(process.env.API_MOCKING) &&
  process.env.API_MOCKING === 'true';

if (isMocking) {
  require('@/mocks');
}

export const StoreProvider = ({ children }: IProvidersProps) => {
  return <Provider store={reduxStore}>{children}</Provider>;
};
