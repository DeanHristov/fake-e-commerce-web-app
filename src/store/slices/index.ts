export * from './apiSlice';
export * from './modalBox';
export * from './products';
export * from './shoppingCart';
export * from './wishList';
