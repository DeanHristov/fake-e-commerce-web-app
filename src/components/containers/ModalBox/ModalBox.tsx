'use client';

import { FC } from 'react';
import { useDispatch, useSelector } from '@/store';
import { closeModal, MODAL_BOX, selectModalBox } from '@/store/slices/modalBox';
import { XMarkIcon } from '@heroicons/react/24/solid';
import ProductDetails from '@/components/containers/ProductDetails';

export interface IModalBoxProps {}

const ModalBox: FC<IModalBoxProps> = ({}) => {
  const dispatch = useDispatch();
  const { isOpen, payload } = useSelector(selectModalBox);

  if (isOpen) {
    return (
      <div className="modal-box-container">
        <div className="content max-w-4xl">
          <div className="relative w-full h-10">
            <XMarkIcon
              data-testid="close-btn"
              className="absolute top-2 right-2 font-medium h-6 w-6 text-eerie-black cursor-pointer transition hover:rotate-90"
              onClick={() => dispatch(closeModal())}
            />
          </div>
          {payload.type === MODAL_BOX.SHOW_PRODUCT && (
            <ProductDetails productId={payload.productId} />
          )}
        </div>
      </div>
    );
  }

  return <div />;
};

export default ModalBox;
