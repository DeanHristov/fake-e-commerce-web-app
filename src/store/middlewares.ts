import { createLogger } from 'redux-logger';
import { apiSlice } from '@/store/slices';

const middlewares = [];

if (process.env.NODE_ENV === 'development') {
  middlewares.push(
    createLogger({
      duration: true,
      timestamp: false,
      collapsed: true,
      colors: {
        title: () => '#139BFE',
        prevState: () => '#1C5FAF',
        action: () => '#149945',
        nextState: () => '#A47104',
        error: () => '#ff0005',
      },
      predicate: () => typeof window !== 'undefined',
    }),
  );
}

middlewares.push(apiSlice.middleware);

export { middlewares };
