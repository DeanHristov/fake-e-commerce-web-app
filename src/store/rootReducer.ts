import { apiSlice, modalBoxSlice } from './slices';

export const reducer = {
  [apiSlice.reducerPath]: apiSlice.reducer,
  [modalBoxSlice.name]: modalBoxSlice.reducer,
};
