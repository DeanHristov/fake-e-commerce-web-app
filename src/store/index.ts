import { configureStore } from '@reduxjs/toolkit';
import {
  type TypedUseSelectorHook,
  useDispatch as useReduxDispatch,
  useSelector as useReduxSelector,
} from 'react-redux'; // @ts-ignore
import { middlewares } from './middlewares';
import { reducer } from './rootReducer'; //@see: https://redux-toolkit.js.org/usage/usage-with-typescript#configurestore

//@see: https://redux-toolkit.js.org/usage/usage-with-typescript#configurestore
export const reduxStore = configureStore({
  reducer,
  middleware: (getDefaultMiddleware) => {
    // @ts-ignore
    return getDefaultMiddleware().concat(middlewares);
  },
});

//@see: https://redux-toolkit.js.org/tutorials/typescript
export type ReduxState = ReturnType<typeof reduxStore.getState>;
export type ReduxDispatch = typeof reduxStore.dispatch;

//@see: https://redux-toolkit.js.org/tutorials/typescript#define-typed-hooks
export const useDispatch = () => useReduxDispatch<ReduxDispatch>();
export const useSelector: TypedUseSelectorHook<ReduxState> = useReduxSelector;
