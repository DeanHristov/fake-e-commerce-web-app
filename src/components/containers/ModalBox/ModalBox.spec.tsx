import { describe } from 'node:test';
import { render } from '@testing-library/react';
import ModalBox from './';
import { useSelector } from '@/store';

const mockSelector = useSelector as jest.Mock;

describe('Containers / ModalBox', () => {
  afterEach(() => mockSelector.mockReset());

  it('Should be able to render the default state', () => {
    mockSelector.mockReturnValue({
      isOpen: true,
      payload: { type: 'something else' },
    });
    const { container } = render(<ModalBox />);

    expect(container).toMatchSnapshot();
  });
});
