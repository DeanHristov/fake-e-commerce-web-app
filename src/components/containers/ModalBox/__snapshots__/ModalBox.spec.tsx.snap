// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be able to render the default state 1`] = `
<div>
  <div
    class="modal-box-container"
  >
    <div
      class="content max-w-4xl"
    >
      <div
        class="relative w-full h-10"
      >
        <svg
          aria-hidden="true"
          class="absolute top-2 right-2 font-medium h-6 w-6 text-eerie-black cursor-pointer transition hover:rotate-90"
          data-testid="close-btn"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M5.47 5.47a.75.75 0 011.06 0L12 10.94l5.47-5.47a.75.75 0 111.06 1.06L13.06 12l5.47 5.47a.75.75 0 11-1.06 1.06L12 13.06l-5.47 5.47a.75.75 0 01-1.06-1.06L10.94 12 5.47 6.53a.75.75 0 010-1.06z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
